operator/internal/controller/pulumi/stack_controller_test.go

```diff
@@ -939,6 +953,21 @@ var _ = Describe("Stack Controller", func() {
 							Expect(err).NotTo(HaveOccurred())
 							ByResyncing()
 						})
+
+						When("the WorkspaceReclaimPolicy is set to Delete", func() {
+							BeforeEach(func(ctx context.Context) {
+								obj.Spec.WorkspaceReclaimPolicy = shared.WorkspaceReclaimDelete
+							})
+							It("reconciles and delete the workspace pod", func(ctx context.Context) {
+								result, err := reconcileF(ctx)
+								Expect(err).NotTo(HaveOccurred())
+								ByMarkingAsReady()
+								By("not requeuing")
+								Expect(result).To(Equal(reconcile.Result{}))
+								By("deleting the Workspace object")
+								Expect(ws.GetName()).To(BeEmpty())
+							})
+						})
```

Comment:
I don't understand why this case is here, in a section that deals with 'ContinueResyncOnCommitMatch'.  I think this block assumes a successful last update.

Reply:
This is nested with the following contexts: `the last update was successful` > `at latest commit` > `ContinueResyncOnCommitMatch` > `a resync is due`. This ensures that we also delete the Workspace pod even if `ContinueResyncOnCommitMatch` is set to true, and the update was successful.